api/api/signals.py

```diff
@@ -1,8 +1,202 @@
 # Copyright (C) 2015, Wazuh Inc.
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+import asyncio
+import logging
+import os
+import ssl
+import uuid
+from functools import wraps
+from typing import AsyncGenerator, Callable
+
+import aiohttp
+import certifi
+from aiohttp import web
+from wazuh.core.cluster.utils import read_cluster_config
+from wazuh.core.configuration import get_ossec_conf
+from wazuh.core.utils import get_utc_now
+
+import wazuh
+from api.constants import INSTALLATION_UID_PATH
+
+CTI_URL = get_ossec_conf(
+    section='global'
+).get('cti_url', 'http://cti:4041')  # This default must be removed once we have the configuratoin in the ossec parser.
+RELEASE_UPDATES_URL = os.path.join(CTI_URL, 'api', 'v1', 'ping')
+ONE_DAY_SLEEP = 60*60*24
+INSTALLATION_UID_KEY = 'installation_uid'
+UPDATE_CHECK_OSSEC_FIELD = 'update_check'
+WAZUH_UID_KEY = 'wazuh-uid'
+WAZUH_TAG_KEY = 'wazuh-tag'
+
+logger = logging.getLogger('wazuh-api')
+
+
+def cancel_signal_handler(func: Callable) -> Callable:
+    """Decorator to handle asyncio.CancelledError for signals coroutines.
+
+    Parameters
+    ----------
+    func : Callable
+        Coroutine to handle.
+
+    Returns
+    -------
+    Callable
+        Wrapped coroutine with exception handled.
+    """
+
+    @wraps(func)
+    async def wrapper(*args, **kwargs):
+        try:
+            await func(*args, **kwargs)
+        except asyncio.CancelledError:
+            pass
+    return wrapper
+
+
+def _get_connector() -> aiohttp.TCPConnector:
+    """Return a TCPConnector with default ssl context.
+
+    Returns
+    -------
+    aiohttp.TCPConnector
+        Instance with default ssl connector.
+    """
+    ssl_context = ssl.create_default_context(cafile=certifi.where())
+    return aiohttp.TCPConnector(ssl=ssl_context)
+
+
+def _get_current_version() -> str:
+    """Return the version of running Wazuh instance
+
+    Returns
+    -------
+    str
+        Wazuh version in format X.Y.Z format.
+    """
+    return wazuh.__version__
+
+
+def _is_running_in_master_node() -> bool:
+    """Determine if cluster is disabled or API is running in a master node.
+
+    Returns
+    -------
+    bool
+        True if API is runing in master node or if cluster is disabled else False.
+    """
+    cluster_config = read_cluster_config()
+
+    return not cluster_config['disabled'] or cluster_config['node_type'] == 'master'
+
+
+def _update_check_is_enabled() -> bool:
+    """Read the ossec.conf and check UPDATE_CHECK_OSSEC_FIELD value.
+
+    Returns
+    -------
+    bool
+        True if UPDATE_CHECK_OSSEC_FIELD is 'yes' or isn't present, else False.
+    """
+    global_configurations = get_ossec_conf(section='global')
+
+    return global_configurations.get(UPDATE_CHECK_OSSEC_FIELD, 'yes') == 'yes'
```

Comment:
Comment: Couldn't find it in the spec, but we should make sure that `true` is also considered as enabled **if the field can contain that value**

Reply:
The validation of the values must be in the ossec parser, here are just reading.